_source/_posts/2025-11-04-user-entitlements-workshop.md

```diff
@@ -0,0 +1,1298 @@
+
+Stop the tunnel and the API using the <kbd>Ctrl</kbd>+<kbd>c</kbd> command in the terminal windows. We'll make some changes to the API that won't automatically reflect in the local tunnel, so we'll get all our entitlements changes made and resynchronize with Okta.
+
+## SCIM schemas and resources 
+
+In the first SCIM workshop, you learned about SCIM's `User` resource and built out operations around the user. You updated only a handful of user properties in the workshop, but SCIM is way more powerful thanks to its superpower—_extensibility_. ✨ User is not the only resource type defined in SCIM.
+
+A `Resource` represents an object SCIM operates on, such as a user or group. SCIM identified core properties each `Resource` must define, such as `id` and a link to the resource's schema definition. From there, a user extends from the core properties and adds attributes specific to the object, such as adding `userName` and their emails. A standard published schema exists for all those user-specific attributes within the SCIM spec. You can continue extending resources as needed to represent new resources, such as another SCIM standard-defined schema for Enterprise User. 
+
+{% img blog/user-entitlements-workshop/resource-class-diagram.svg alt:"Class diagram representing core Resource properties, User class extending from core Resource adds username and emails properties. The Enterprise User class extends from User adds department and costCenter properties. Group class extends from core Resource and adds displayName and members properties. Other class extends from core Resource demonstrating new resource representations." width:"800" %}{: .center-image }
+
+{% comment %}
+Tweak the diagram on https://mermaid.live/ with the following content
+classDiagram
+    Resource <|-- User
+    Resource <|-- Group
+    User <|-- Enterprise User
+    Resource <|-- Other
+    Resource : id
+    Resource: schemas
+    User : userName
+    User : list of emails
+    Group: list of members
+    EnterpriseUser : department
+    EnterpriseUser : costCenter
+{% endcomment %}
+
+What's an example resource other than a user or group? If you said "role" or an "entitlement," you're correct! Those resource types must have an `id` and `schemas`. Here, Okta used SCIM's extensibility to define a new resource type.
+
+{% img blog/user-entitlements-workshop/resource-role-class-diagram.svg alt:"Class diagram representing core Resource properties. The User, Group, OktaRole, and Other class extends from core Resource." width:"800" %}{: .center-image }
+
+{% comment %}
+Tweak the diagram on https://mermaid.live/ with the following content
+classDiagram
+    Resource <|-- User
+    Resource <|-- Group
+    Resource <|-- OktaRole
+    Resource <|-- Other
+    Resource : id
+    Resource: schemas
+{% endcomment %}
+
+Okta defines a schema for the `Role` representation. We can use the schema to ensure we conform to the definition.
```

Comment:
```suggestion
Okta defines a [schema](https://developer.okta.com/docs/guides/scim-with-entitlements/main/#role-example) for the `Role` representation. We can use the schema to ensure we conform to the definition.
```

maybe there's a better link? if so, it has dismal SEO. Either way, showing the schema we mean is nice.